rln-research/membership-merkle-tree-storage-evaluation.md

```diff
@@ -0,0 +1,107 @@
+# Waku-RLN-Relay: Evaluating Storage Overhead of Membership Merkle Tree   
+## Introduction
+
+In the current design of the Waku-RLN-Relay protocol, the membership Merkle tree (MT) is locally persisted by each peer. This incurs some storage overhead which may or may not fit resource-limited devices. This overhead also depends on the algorithm used to represent the Merkle tree. So far, we are aware of two algorithms namely, Full Merkle tree (FMT) and Optimal Merkle tree (OMT), and in this post, we would like
```

Comment:
Is there any difference in requirements for "each peer" vs nodes that only validate?

Reply:
`nodes that only validate` are the relayers, however, the knowledge of Merkle tree is required for both publishers and relayers i.e., `each peer`. Does that address your question?

Reply:
I guess it is more about clarifying in the text that it is (currently?) a requirement for both publishers and validators/(relayers). I think one could imagine publishing and validation requirements differing in the future.

Reply:
Sure, done in https://github.com/vacp2p/research/pull/132/commits/708292428696240b97520a31dd03b3f0094d6cd5